### ISSUE TITLE
Reduce expensive calls to fileChannel.size() in <AbstractWire>.readFirstHeader()

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Wire#178]
readFirstHeader() was calling ( in a loop ) realCapacity(), realCapacity() should only have to be called once and checked if its len is greater than 4. for example change to this

```
try {
    boolean hasAtLeast4 = false;
    for (; ; ) {
        if (hasAtLeast4 || bytes.realCapacity() >= 4) {
            hasAtLeast4 = true;
            header = bytes.readVolatileInt(0L);
            if (isReady(header))
                break;
        }
        acquireTimedParser().pause(timeout, timeUnit);
    }
```

**Comment by RobAustin** [comment]
once recovering the corrupt header we still have to recreate it

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
